lib/improver/tests/weighted_blend/test_TriangularWeightedBlendAcrossAdjacentPoints.py

```diff
@@ -0,0 +1,227 @@
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the
+   weighted_blend.WeightedBlendAcrossAdjacentPoints plugin."""
+
+import unittest
+
+from cf_units import Unit
+
+from iris.coords import DimCoord
+from iris.cube import Cube
+from iris.tests import IrisTest
+
+import numpy as np
+
+from improver.weighted_blend import TriangularWeightedBlendAcrossAdjacentPoints
+
+
+class Test__repr__(IrisTest):
+
+    """Test the __repr__ method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        result = str(TriangularWeightedBlendAcrossAdjacentPoints('time',
+                                                                 3.0, 'hours'))
+        msg = ('<TriangularWeightedBlendAcrossAdjacentPoints:'
+               ' coord = time, width = 3.00,'
+               ' parameter_units = hours>')
+        self.assertEqual(result, msg)
+
+
+class Test__init__(IrisTest):
+
+    """Test the __init__ method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        plugin = TriangularWeightedBlendAcrossAdjacentPoints('time',
+                                                             3.0, 'hours')
+        expected_coord = "time"
+        expected_width = 3.0
+        expected_parameter_units = "hours"
+        self.assertEqual(plugin.coord, expected_coord)
+        self.assertEqual(plugin.width, expected_width)
+        self.assertEqual(plugin.parameter_units, expected_parameter_units)
+
+
+class Test_correct_collapsed_coordinates(IrisTest):
+
+    """Test the correct_collapsed_coordinates method"""
+
+    def setUp(self):
+        """Set up a test orig_cube, new_cube and plugin instance."""
+        data = np.zeros((2, 2, 2))
+
+        orig_cube = Cube(data, units="m",
```

Comment:
Is this section significantly different from other set_up_cube sections? Are none of those suitable, as it would be best to not continue setting up these test cubes, if necessary? See #50 .